### DISCUSSION TITLE
How do I get Ctrl-Backspace to delete a word?

### DISCUSSION THREAD

**Discussion by wild-endeavor** [wezterm/wezterm#3983] [General]
I just can't get used to Ctrl-w.

This is on a Mac, running a recent version of wezterm.

Thank you.

**Comment by wez** [comment]
You could assign Ctrl-Backspace to Ctrl-W if you like.

Take a look at https://wezfurlong.org/wezterm/config/lua/keyassignment/SendKey.html

**Comment by wild-endeavor** [comment]
Thanks!  Added 

```
{ key = 'Backspace', mods = 'CTRL', action = act.SendKey {key = 'w', mods = 'CTRL'} },
```
for now.  works!  will see if any mishaps occur in lvim.  this is the right way of doing this?  like this is the right layer?  i get confused between `backward-kill-word`, `^H`, and ctrl-w.  i feel like i've used every combination of everything in the past.

**Reply by wez** [reply]
It's up to you really. If you only want ctrl-backspace to work that way in your shell, then it may be better to define that mapping for your shell using its key binding facilities.

**Reply by sondhg** [reply]
> { key = 'Backspace', mods = 'CTRL', action = act.SendKey {key = 'w', mods = 'CTRL'} },

This works great! But when inside search mode via `Ctrl+Shift+F`, do you know how to delete a whole word, either backward or forward? `Ctrl+Backspace` and `Ctrl+w` can't delete backwards, and `Ctrl+Delete` can't delete forward, as far as I know

**Comment by RheaAyase** [comment]
`Ctrl+W` doesn't really "delete word" correctly, it doesn't separate words correctly at all. It's more like "delete the entire space delimited sentence" i.e. when you ctrl+W/backspace you want to delete one directory in a path - not the entire path.

In order to delete word backwards respecting path separation you can add to your `~/.inputrc` (readline behaviour)
```
"\C-w": unix-filename-rubout
```
(Plus the ctrl+backspace keybind in wezterm?)

Alternative solution if the above isn't suitable:
```
  {
    key = 'Backspace',
    mods = 'CTRL',
    action = wezterm.action.Multiple {
      wezterm.action.SendKey { key = 'b', mods = 'ALT' },
      wezterm.action.SendKey { key = 'Delete', mods = 'CTRL' },
    }
  }
```
Move left by word, then delete word, It has the caveat of not being able to delete word left if you're in the middle of it, cuz it would delete the whole thing. Similarly with some other cases if you're trying to delete in the middle of the path and not from the end. However I think that's a rare use case, most of the time you just want to delete whatever filename is at the end while keeping the rest of the path, so this will do the trick.
